docs/how-to-guides/shared-libraries.rst

```diff
@@ -0,0 +1,253 @@
+.. SPDX-FileCopyrightText: 2024 The meson-python developers
+..
+.. SPDX-License-Identifier: MIT
+
+.. _shared-libraries:
+
+**********************
+Using shared libraries
+**********************
+
+Python projects may build shared libraries as part of their project, or link
+with shared libraries from a dependency. This tends to be a common source of
+issues, hence this page aims to explain how to include shared libraries in
+wheels, any limitations and gotchas, and how support is implemented in
+``meson-python`` under the hood.
+
+We distinguish between *internal* shared libraries, i.e. they're built as part
+of the build executed by ``meson-python``, and *external* shared libraries that
+are only linked against from targets (usually Python extension modules) built
+by ``meson-python``. For internal shared libraries, we also distinguish whether
+the shared library is being installed to its default location (i.e. ``libdir``,
+usually something like ``<prefix>/lib/``) or to a location in ``site-packages``
```

Comment:
I would substitute the parenthetical statement with "(typically `/usr/local/lib` on Unix-like systems, and `?` on Windows)". I don't know what the default prefix is on Windows and I'm not sure the default prefix is `/usr/local`.

Reply:
Yes, `/usr/local/lib` and `C:\\lib`, see https://mesonbuild.com/Builtin-options.html#universal-options.